backend/executor.go

```diff
@@ -135,9 +141,8 @@ func (executor *grpcExecutor) ExecuteOrchestrator(ctx context.Context, iid api.I
 // ExecuteActivity implements Executor
 func (executor *grpcExecutor) ExecuteActivity(ctx context.Context, iid api.InstanceID, e *protos.HistoryEvent) (*protos.HistoryEvent, error) {
 	key := getActivityExecutionKey(string(iid), e.EventId)
-	result := &activityExecutionResult{complete: make(chan interface{})}
+	result := &activityExecutionResult{complete: make(chan struct{})}
 	executor.pendingActivities.Store(key, result)
-	defer executor.pendingActivities.Delete(key)
 
```

Comment:
Why are we removing the calls to delete these keys? Won't that result in a memory leak?

Reply:
They will be deleted by `CompleteActivityTask`. Additionally, if the `GetWorkItems` strema is closed while the operation is in progress, the cleanup logic added to `GetWorkItems` will delete any "leftover" item in this map

Reply:
Is there any reason for moving the delete operation from here to other places? If I understand corretly, the delete then needs to be added in `CompleteActivityTask`, in the defer block in `GetWorkItems` and potentially `Shutdown` as well. Instead if we just keep it here, it would handle all scenarios. Also looks more intuitive to me(We would still need to close the `complete` channel in getWorkItems and `Shutdown`)

Reply:
Good question. I wrote this a week ago and now I cannot remember why, even after reading the code :( It may work

Reply:
Then this may help to avoid closing a closed channel, as we delete it in time.

Reply:
It's impossible currently for `close()` to be called on the same channel twice

Reply:
yes, I mean currently implementation will help avoid closing a closed channel, but with `defer executor.pendingActivities.Delete(key)`, we can potentially close a closed channel if the stream is closed and then the shutdown is called.